src/lib/Scenes/Artwork/Artwork.tsx

```diff
@@ -1,47 +1,126 @@
 import { Artwork_artwork } from "__generated__/Artwork_artwork.graphql"
 import { ArtworkQuery } from "__generated__/ArtworkQuery.graphql"
 import Separator from "lib/Components/Separator"
 import { defaultEnvironment } from "lib/relay/createEnvironment"
 import renderWithLoadProgress from "lib/utils/renderWithLoadProgress"
 import React from "react"
-import { Dimensions, ScrollView } from "react-native"
+import { Dimensions, FlatList } from "react-native"
 import { createFragmentContainer, graphql, QueryRenderer } from "react-relay"
 import { AboutArtistFragmentContainer as AboutArtist } from "./Components/AboutArtist"
-import { AboutWorkFragmentContainer as AboutWork } from "./Components/AboutWork"
-import { ArtworkActionsFragmentContainer as ArtworkActions } from "./Components/ArtworkActions"
-import { ArtworkAvailabilityFragmentContainer as ArtworkAvailability } from "./Components/ArtworkAvailability"
+import { AboutWork } from "./Components/AboutWork"
+import { ArtworkAvailability } from "./Components/ArtworkAvailability"
 import { ArtworkDetailsFragmentContainer as ArtworkDetails } from "./Components/ArtworkDetails"
+import { ArtworkHeader } from "./Components/ArtworkHeader"
 import { ArtworkTombstoneFragmentContainer as ArtworkTombstone } from "./Components/ArtworkTombstone"
-import { ImageCarouselFragmentContainer as ImageCarousel } from "./Components/ImageCarousel/ImageCarousel"
 import { OtherWorksFragmentContainer as OtherWorks } from "./Components/OtherWorks"
-import { SellerInfoFragmentContainer as SellerInfo } from "./Components/SellerInfo"
+import { SellerInfo } from "./Components/SellerInfo"
 
 interface Props {
   artwork: Artwork_artwork
 }
 
+interface State {
+  sections: Array<{
+    type: "tombstone" | "availability" | "sellerInfo" | "aboutWork" | "details" | "aboutArtist" | "otherWorks"
+    data: any
+  }>
+}
+
 export class Artwork extends React.Component<Props> {
-  render() {
+  state: State = {
+    sections: [],
+  }
+
+  componentDidMount() {
+    const { artwork } = this.props
+    const sections = []
+
+    sections.push({
+      type: "header",
+    })
+
+    sections.push({
+      type: "tombstone",
+    })
+
+    if (artwork.availability) {
+      sections.push({
+        type: "availability",
+        data: artwork.availability,
+      })
+    }
+
+    if (artwork.partner && artwork.partner.name) {
+      sections.push({
+        type: "sellerInfo",
+        data: artwork.partner.name,
+      })
+    }
+
+    if (artwork.description || artwork.additional_information) {
+      sections.push({
+        type: "aboutWork",
+        data: { description: artwork.description, additionalInformation: artwork.additional_information },
+      })
+    }
+
+    sections.push({
+      type: "details",
+    })
+
+    sections.push({
+      type: "aboutArtist",
+    })
+
+    sections.push({
+      type: "otherWorks",
+    })
+
+    this.setState({ sections })
+  }
+
+  renderItem = ({ item: { data, type } }) => {
```

Comment:
Add the appropriate ‘item’ typing from the RN types.

Reply:
Btw, if you were to find the right data from your props here inline instead of first making a list of sections, then you also don’t need to type all of the `data` fields. Additionally the list of sections could just be a static (module level) constant array, no need to use state for that at all (which it isn’t really anyways, it’s just a different view on your props).

Reply:
Ah, my bad, when I reviewed this earlier from my phone the context got lost on me a bit. So seeing as there are some conditionals, the list of sections needs to indeed be determined based on the data, but the actual data for each section doesn’t need to be captured here. (Will elaborate in a pairing session.)